packages/cra-template-typescript/template/src/service-worker.ts

```diff
@@ -0,0 +1,50 @@
+// This service worker can be customized!
+// See https://developers.google.com/web/tools/workbox/modules
+// for the list of available Workbox modules, or add any other
+// code you'd like.
+// You can also remove this file if you'd prefer not to use a
+// service worker, and the Workbox build step will be skipped.
+
+declare const self: ServiceWorkerGlobalScope;
```

Comment:
A reference to the `webworker` TypeScript library is needed for this.

I could add

```
<reference lib="webworker" />
```

but that's probably not encouraged; is there a standard way in `c-r-a` to add a library reference to the user's `tsconfig`?

Reply:
See https://github.com/microsoft/TypeScript/issues/11781

Reply:
I've added a reference to this file. We _could_ add to the tsconfig `lib` array, but i figure we probably want to restrict those types to only this file?

Reply:
Resolving, as the file has moved to https://github.com/cra-template/pwa/blob/master/packages/cra-template-pwa-typescript/template/src/service-worker.ts